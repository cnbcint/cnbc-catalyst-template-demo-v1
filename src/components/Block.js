import React, { useContext } from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { breakpoints, getMargin } from "../helpers/Layout";
import { LayoutContext } from "./Main";

const Container = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  padding-top: ${getMargin(3)};
  padding-bottom: ${getMargin(3)};
  background-image: ${(props) =>
    props.backgroundImage ? "url(" + props.backgroundImage + ")" : "none"};
  @media (min-width: ${breakpoints.mobile}) {
    padding-top: ${getMargin(5)};
    padding-bottom: ${getMargin(5)};
  }
`;
const Content = styled.div`
  width: calc(100% - ${getMargin(2)});
  display: flex;
  flex-direction: column;
  max-width: ${(props) => props.maxWidth};
`;
/**
 *
 * Block is the main wrapper component for this template.
 * This element enables to have consistent spacing between sections using the theme functionality to define a central column
 * You can include most of the components
 */
export default function Block(props) {
  const layout = useContext(LayoutContext);
  return (
    <Container
      className={props.className}
      style={props.style}
      backgroundImage={props.backgroundImage}
    >
      <Content maxWidth={layout.maxWidth}>{props.children}</Content>
    </Container>
  );
}

Block.defaultProps = {
  style: {},
  className: "",
  backgroundImage: null,
};

Block.propTypes = {
  className: PropTypes.string,
  backgroundImage: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array,
    PropTypes.element,
  ]).isRequired,
  /**
  React CSS format to override rules has required by design
  */
  style: PropTypes.object,
};
