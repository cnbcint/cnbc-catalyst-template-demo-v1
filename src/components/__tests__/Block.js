import React from "react";
import Block from "../Block";
import { render } from "@testing-library/react";
import "jest-styled-components";

const Component = () => (
  <Block
    className="testClass"
    style={{ backgroundColor: "rgb(255, 255, 255)" }}
  >
    Testing Content
  </Block>
);

beforeAll(() => {
  window.IntersectionObserver = class {
    observe() {
      return null;
    }
    disconnect() {}
  };
});

describe("Block", () => {
  it("Renders correctly", () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Renders the correct content", () => {
    const { container } = render(<Component />);
    expect(container.firstChild.textContent).toMatch("Testing Content");
  });

  it("Passes the class name", () => {
    const { container } = render(<Component />);
    expect(container.firstChild.classList.contains("testClass")).toBe(true);
  });

  it("Passes the style", () => {
    const { container } = render(<Component />);
    expect(container.firstChild.style["background-color"]).toBe(
      "rgb(255, 255, 255)"
    );
  });
});

afterAll(() => {
  window.IntersectionObserver = null;
});
