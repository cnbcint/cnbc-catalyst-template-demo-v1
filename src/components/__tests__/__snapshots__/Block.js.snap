// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block Renders correctly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-top: 3rem;
  padding-bottom: 3rem;
  background-image: none;
}

.c1 {
  width: calc(100% - 2rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 500px;
}

@media (min-width:768px) {
  .c0 {
    padding-top: 5rem;
    padding-bottom: 5rem;
  }
}

<div
  class="c0 testClass"
  style="background-color: rgb(255, 255, 255);"
>
  <div
    class="c1"
  >
    Testing Content
  </div>
</div>
`;
